packages/web/src/components/splitView.tsx

```diff
@@ -38,8 +38,8 @@ export const SplitView: React.FC<React.PropsWithChildren<SplitViewProps>> = ({
   settingName,
   children
 }) => {
-  const [hSize, setHSize] = useSetting<number>(settingName ? settingName + '.' + orientation + ':size' : undefined, Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio);
-  const [vSize, setVSize] = useSetting<number>(settingName ? settingName + '.' + orientation + ':size' : undefined, Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio);
+  const [hSize, setHSize] = useSetting<number>((settingName ?? 'dont-persist') + '.' + orientation + ':size', Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio, undefined, !settingName);
```

Comment:
I kind of liked the previous api with passing `undefined` more. Can we revert that part?

Reply:
Not really, because we need a way of attributing the `useSetting` call with a setting name to store it inside `Settings#sessionSettings`. If you don't want this saved, we could make it a `useState` call instead?

Reply:
Implemented in https://github.com/microsoft/playwright/pull/31911/commits/00a1574737802e74464e116582b272eaf6b9f265.